hack/e2e.go

```diff
@@ -337,19 +352,180 @@ func Build() error {
 	return nil
 }
 
-func TearDown() error {
-	return finishRunning("teardown", exec.Command("./hack/e2e-internal/e2e-down.sh"))
+type deployer interface {
+	Up() error
+	IsUp() error
+	SetupKubecfg() error
+	Down() error
+}
+
+func getDeployer() (deployer, error) {
+	switch *deployment {
+	case "bash":
+		return bash{}, nil
+	case "kops":
+		return NewKops()
+	default:
+		return nil, fmt.Errorf("Unknown deployment strategy %q", *deployment)
+	}
 }
 
-func Up() error {
+type bash struct{}
+
+func (b bash) Up() error {
 	return finishRunning("up", exec.Command("./hack/e2e-internal/e2e-up.sh"))
 }
 
-// Is the e2e cluster up?
-func IsUp() error {
+func (b bash) IsUp() error {
 	return finishRunning("get status", exec.Command("./hack/e2e-internal/e2e-status.sh"))
 }
 
+func (b bash) SetupKubecfg() error {
+	return nil
+}
+
+func (b bash) Down() error {
+	return finishRunning("teardown", exec.Command("./hack/e2e-internal/e2e-down.sh"))
+}
+
+type kops struct {
+	path    string
+	zones   []string
+	nodes   int
+	cluster string
+	kubecfg string
+}
+
+func NewKops() (*kops, error) {
+	if *kopsPath == "" {
+		return nil, fmt.Errorf("--kops must be set to a valid binary path for kops deployment.")
+	}
+	if *kopsCluster == "" {
+		return nil, fmt.Errorf("--kops-cluster must be set to a valid cluster name for kops deployment.")
+	}
+	if *kopsState == "" {
+		return nil, fmt.Errorf("--kops-state must be set to a valid S3 path for kops deployment.")
+	}
+	if err := os.Setenv("KOPS_STATE_STORE", *kopsState); err != nil {
+		return nil, err
+	}
+	f, err := ioutil.TempFile("", "kops-kubecfg")
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
```

Comment:
The ioutil.TempFile semantics leave the file until you unlink it, so this should be fine.